src/functionaltests/java/com/ericsson/ei/utils/JenkinsManager.java

```diff
@@ -0,0 +1,226 @@
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE).addParam("token", token)
+                .setEndpoint(endpoint);
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        success = response.getStatusCode() == HttpStatus.CREATED;
+        return success;
+    }
+
+    /**
+     * This function recieves a jenkins job name and returns weather the job has
+     * been triggered at least once
+     *
+     * @param jobName
+     * @return
+     * @throws URISyntaxException
+     */
+    public boolean jobHasBeenTriggered(String jobName) throws URISyntaxException {
+        boolean triggered = false;
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        String endpoint = "/job/" + jobName + "/1/api/json";
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE)
+                .setEndpoint(endpoint);
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        triggered = response.getStatusCode() == HttpStatus.OK;
+        return triggered;
+    }
+
+    /**
+     *
+     * This function recieves a jenkins job name and deletes that job from the
+     * jenkins system
+     *
+     * @param jobName
+     * @return
+     * @throws URISyntaxException
+     */
+    public boolean deleteJob(String jobName) throws URISyntaxException {
+        boolean deleted = false;
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        String endpoint = "/job/" + jobName + "/doDelete";
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE).addHeader("Jenkins-Crumb", crumb)
+                .setEndpoint(endpoint);
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        deleted = response.getStatusCode() == HttpStatus.FOUND;
+        return deleted;
+    }
+
+    /**
+     * Gets the jenkins crumb if any
+     *
+     * @return
+     * @throws URISyntaxException
+     */
+    private String getCrumb() throws URISyntaxException {
+        String crumb = "";
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE).setEndpoint("/crumbIssuer/api/json");
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        boolean success = response.getStatusCode() == HttpStatus.OK;
+
+        if (success) {
+            JSONObject jsonObj = new JSONObject(response.getBody());
```

Comment:
Isn't this deprecated? I might be wrong but i have ghost in my head telling me it is 👻

Reply:
JSONArray is :) JSONObject is still ok.

Reply:
Oh great :)